src/plugins/intel_gpu/src/graph/graph_optimizer/basic_memory_dependencies.cpp

```diff
@@ -36,14 +36,13 @@ void basic_memory_dependencies::run(program& p) {
             add_memory_dependency(it, node);
         }
 
-        if (node->is_type<convolution>() && node->get_preferred_impl_type() == impl_types::onednn) {
-            auto& conv = node->as<convolution>();
+        if (node->get_preferred_impl_type() == impl_types::onednn) {
             bool can_reuse_eltwise_mem = false;
             size_t eltw_dep = 0;
 
-            for (auto& fused_op : conv.get_fused_primitives()) {
+            for (auto& fused_op : node->get_fused_primitives()) {
```

Comment:
This is too general. For example, `pooling` does not support sum post operation.